From 344c333751f954c82d7ef3f67a96681be592e10b Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Mon, 15 Sep 2014 17:00:32 +0800
Subject: [PATCH 254/255] gianfar: always start the TBI phy auto negotiation

The external SGMII phy may be reset during probe, so we should
always start the TBI phy auto negotiation instead of the link
status.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/freescale/gianfar.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/net/ethernet/freescale/gianfar.c b/drivers/net/ethernet/freescale/gianfar.c
index d8f3469..150f921 100644
--- a/drivers/net/ethernet/freescale/gianfar.c
+++ b/drivers/net/ethernet/freescale/gianfar.c
@@ -1740,14 +1740,6 @@ static void gfar_configure_serdes(struct net_device *dev)
 		return;
 	}
 
-	/* If the link is already up, we must already be ok, and don't need to
-	 * configure and reset the TBI<->SerDes link.  Maybe U-Boot configured
-	 * everything for us?  Resetting it takes the link down and requires
-	 * several seconds for it to come back.
-	 */
-	if (phy_read(tbiphy, MII_BMSR) & BMSR_LSTATUS)
-		return;
-
 	/* Single clk mode, mii mode off(for serdes communication) */
 	phy_write(tbiphy, MII_TBICON, TBICON_CLK_SELECT);
 
-- 
2.0.2

